### PR TITLE
Feat 5 liked videos

### PR DESCRIPTION
**Pull request by SinchanaMS**
This request adds the following features:

1. The like feature: Users can like and unlike the videos, both from the explore page as well as from the single video page.
2. Single video page: On clicking the thumbnail, the user is redirected to the single video page.

### Preview:
1. The like feature on Explore page:

image

2. Single video page:

image
image

### CODE REVIEW

File: src/router/utils/HelperFunctions.jsx [hunk 1]
```diff
@@ -50,3 +51,50 @@ export const removeFromWatchlist = async (video, videoDispatch) => {
     console.log(error);
   }
 };
+
```
Comment (ShrutiLakhani) [inline src/router/utils/HelperFunctions.jsx:54]: Suggestion: Instead of Console logging the error, showing it to the user would be better.

**Review by ShrutiLakhani — APPROVED** [review]
Well structured. Clean Code. Great Job  👍

File: src/styles/video.css [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+.suggestions {
+  overflow-x: auto;
+  margin-bottom: 2rem;
+}
+
+.videocard .video-card {
+  width: 15rem;
+  margin: 8px;
+  height: 13.5rem;
+  min-height: 13.5rem;
```
Comment (pankaj-prog) [inline src/styles/video.css]: only height:13.5rem will do the same, why did u specify additional min-height as well

File: src/styles/video.css [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+.react-player-card {
+  width: 1536px;
+  height: 1024px;
+}
+
+.player-ctas {
+  display: flex;
+  gap: 1rem;
+  margin-top: 1rem;
```
Comment (pankaj-prog) [inline src/styles/video.css:52]: you are repeating this at various places, you can make a class name, gutter-bottom or gutter-top and use it wherever you need.

File: src/pages/VideoPage/VideoPage.jsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            </div>
+          </div>
+        </div>
+      </div>
+
+      <div className="suggestions">
+        <h3>View similar</h3>
+        <div className="videos">
+          {suggestedList.map((video) => (
+            <ul key={video?._id}>
```
Comment (pankaj-prog) [inline src/pages/VideoPage/VideoPage.jsx]: here you are creating multiple lists with a single item only, instead, you should be wrapping the map function inside ul element and then return multiple li items

File: src/pages/VideoPage/VideoPage.jsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        </div>
+      </div>
+
+      <div className="suggestions">
+        <h3>Watch more</h3>
+        <div className="videos">
+          {randomVideos.map((video) => (
+            <ul key={video?._id}>
+              <li className="videocard">
+                <VideoCard video={video} key={video?._id} />
```
Comment (pankaj-prog) [inline src/pages/VideoPage/VideoPage.jsx]: same here, wrap map inside ul

**Review by pankaj-prog — APPROVED** [review]

File: src/contexts/VideoContext.jsx [hunk 6]
```diff
@@ -1,18 +1,47 @@
...
   history: [],
 };
+
 const VideoProvider = ({ children }) => {
   const [videoData, videoDispatch] = useReducer(videoReducer, initialData);
+  const [videoList, setVideoList] = useState([]);
+  const [showOptions, setShowOptions] = useState(false);
+
+  useEffect(() => {
+    const fetchVideos = async () => {
```
Comment (ritikaawasthy) [inline src/contexts/VideoContext.jsx:31]: you could shift all the API calls to a different files will make the code modular

**Review by ritikaawasthy — COMMENTED** [review]
Code looks good, its clean and readable.

File: src/router/utils/HelperFunctions.jsx (hunk 1 above)
Comment (SinchanaMS) [reply src/router/utils/HelperFunctions.jsx:54]: Yes, I'm yet to add the toasts. Thank you!

File: src/pages/VideoPage/VideoPage.jsx (hunk 4 above)
Comment (SinchanaMS) [reply src/pages/VideoPage/VideoPage.jsx]: Oh right! Thanks for pointing it out, I missed this.

File: src/styles/video.css (hunk 2 above)
Comment (SinchanaMS) [reply src/styles/video.css]: I missed removing that line, will do it. Thank you!

**Comment by SinchanaMS** [comment]
Thank you for the reviews. Made the necessary changes. Merging to dev.

**Pull request merged**